abstract_commands.xml

```diff
@@ -2,13 +2,13 @@
     <register name="Access Register">
         \begin{steps}{Perform the following sequence of operations:}
         \item Halt the hart, if \Fprehalt is set. If the hart is already
-            handled, the entire command fails.
+            halted, the entire command fails.
         \item Execute the Program Buffer, if \Fpreexec is set.
         \item Copy data from the register specified by \Fregno into {\tt data}, if \Fwrite is clear.
         \item Copy data from {\tt data} into the register specified by \Fregno, if \Fwrite is set.
         \item Execute the Program Buffer, if \Fpostexec is set.
-        \item Resume the hart, if \Fpostresume is set. If the hart is already
-            running, the entire command fails.
+        \item Resume the hart, if \Fpostresume is set. If the hart is already 
+        running and \Fprehalt is not set, the entire command fails.
```

Comment:
My reading of the list was that it was a sequential explanation of behaviour ("Perform the following sequence of actions"). 

The way I understand it, the failure case for line 6 is only encountered if abstract commands can be used without halting the core, and postresume was set but a command was used without halting the core. If the hart was already running and prehalt was set, that failure mode is already described in line 1.

I think currently the spec is giving a simple sequential model of things, and it's left as an exercise for the reader to map to hardware (as you suggest, there are advantages to determining if the command is valid or not immediately, and an implementer might choose to do this). I don't have a strong view on which way of describing behaviour should be preferred.